pkg/term/term_windows.go

```diff
@@ -29,53 +31,98 @@ func StdStreams() (stdIn io.ReadCloser, stdOut, stdErr io.Writer) {
 		return os.Stdin, os.Stdout, os.Stderr
 	case os.Getenv("MSYSTEM") != "":
 		// MSYS (mingw) does not emulate ANSI well.
-		return winconsole.WinConsoleStreams()
+		return windows.ConsoleStreams()
 	default:
-		return winconsole.WinConsoleStreams()
+		return windows.ConsoleStreams()
 	}
 }
 
 // GetFdInfo returns file descriptor and bool indicating whether the file is a terminal.
 func GetFdInfo(in interface{}) (uintptr, bool) {
-	return winconsole.GetHandleInfo(in)
+	return windows.GetHandleInfo(in)
 }
 
 // GetWinsize retrieves the window size of the terminal connected to the passed file descriptor.
 func GetWinsize(fd uintptr) (*Winsize, error) {
-	info, err := winconsole.GetConsoleScreenBufferInfo(fd)
+
+	info, err := windows.GetConsoleScreenBufferInfo(fd)
+	if err != nil {
+		return nil, err
+	}
+
+	font, err := windows.GetCurrentConsoleFont(fd)
 	if err != nil {
 		return nil, err
 	}
 
-	// TODO(azlinux): Set the pixel width / height of the console (currently unused by any caller)
-	return &Winsize{
+	winsize := &Winsize{
 		Width:  uint16(info.Window.Right - info.Window.Left + 1),
 		Height: uint16(info.Window.Bottom - info.Window.Top + 1),
-		x:      0,
-		y:      0}, nil
+		x:      uint16(font.FontSize.X),
+		y:      uint16(font.FontSize.Y)}
+
+	return winsize, nil
 }
 
 // SetWinsize sets the size of the given terminal connected to the passed file descriptor.
 func SetWinsize(fd uintptr, ws *Winsize) error {
-	// TODO(azlinux): Implement SetWinsize
-	logrus.Debugf("[windows] SetWinsize: WARNING -- Unsupported method invoked")
-	return nil
+
+	// Ensure the requested dimensions are no larger than the maximum window size
+	info, err := windows.GetConsoleScreenBufferInfo(fd)
+	if err != nil {
+		return err
+	}
+
+	if ws.Width == 0 || ws.Height == 0 || ws.Width > uint16(info.MaximumWindowSize.X) || ws.Height > uint16(info.MaximumWindowSize.Y) {
+		return fmt.Errorf("Illegal window size: (%v,%v) -- Maximum allow: (%v,%v)",
+			ws.Width, ws.Height, info.MaximumWindowSize.X, info.MaximumWindowSize.Y)
+	}
+
+	// Narrow the sizes to that used by Windows
+	width := windows.SHORT(ws.Width)
+	height := windows.SHORT(ws.Height)
+
+	// Set the dimensions while ensuring they remain within the bounds of the backing console buffer
+	// -- Shrinking will always succeed. Growing may push the edges past the buffer boundary. When that occurs,
+	//    shift the upper left just enough to keep the new window within the buffer.
+	rect := info.Window
+	if width < rect.Right-rect.Left+1 {
+		rect.Right = rect.Left + width - 1
```

Comment:
might as well write:
```
rect := info.Window
rect.Right = rect.Left + width - 1
if rect.Right >= info.Size.X {
    rect.Left = info.Size.X - width
    rect.Right = info.Size.X - 1
}
```

Hopefully `if rect.Right >= info.Size.X { ...` is equivalent to `if width > rect.Right-rect.Left+1 { rect.Right = rect.Left + width - 1; if rect.Right >= info.Size.X { ...`.